oci/oci.go

```diff
@@ -689,7 +689,18 @@ func (r *Runtime) StopContainer(ctx context.Context, c *Container, timeout int64
 
 	if timeout > 0 {
 		if err := utils.ExecCmdWithStdStreams(os.Stdin, os.Stdout, os.Stderr, r.Path(c), "kill", c.id, c.GetStopSignal()); err != nil {
-			return fmt.Errorf("failed to stop container %s, %v", c.id, err)
+			process, perr := findprocess.FindProcess(c.state.Pid)
+			if perr == findprocess.ErrNotFound {
+				c.state.Finished = time.Now()
+				return nil
+			}
+			if perr == nil {
+				err = process.Release()
+				if err != nil {
+					logrus.Warnf("failed to release process %d for container %q: %v", c.state.Pid, c.ID(), err)
+				}
+			}
+			return fmt.Errorf("failed to stop container: %s, err: %v, findprocess err: %v", c.id, err, perr)
```

Comment:
While we're touching this code, I'd rather have this message (and the similar one in the following hunk) keep the container ID to the let of the colon (like we do in many other places, `git grep 'container %q`).  Maybe:

```go
return fmt.Errorf("failed to stop container %q: %v (findprocess err: %v)", c.id, err, perr)
```